src/main/java/capstone/mfslbackend/service/TransactionService.java

```diff
@@ -55,9 +68,28 @@ public Transaction createTransaction(Long fantasyTeamId, Long incomingPlayerId,
         }
 
         Player playerIn = playerService.getPlayerById(incomingPlayerId);
-        transaction.setPlayerIn(playerIn);
+        Player outgoingPlayer = playerService.getPlayerById(outgoingPlayerId);
+
+        if (transaction.getReceivingFantasyTeam() == null) { //it means it is a free agent
+            if (validTransaction(proposingFantasyTeam, playerIn, outgoingPlayer)) { //if valid, transaction is accepted
+                String replacement = proposingFantasyTeam.getPlayerIdsInOrder().replace(outgoingPlayerId.toString(), incomingPlayerId.toString());
+                proposingFantasyTeam.setPlayerIdsInOrder(replacement);
 
-        transaction.setStatus(TransactionStatus.PROPOSED);
+                transaction.setPlayerIn(playerIn);
+                Set<Player> currentPlayers = proposingFantasyTeam.getPlayers();
+                currentPlayers.remove(outgoingPlayer);
+                currentPlayers.add(playerIn);
+                proposingFantasyTeam.setPlayers(currentPlayers);
+                transaction.setStatus(TransactionStatus.ACCEPTED);
+
+            } else {
+                transaction.setStatus(TransactionStatus.REJECTED);
+            }
+
+        } else {
+            transaction.setPlayerIn(playerIn);
```

Comment:
you should move setPlayerIn outside all the ifs, because you need to do it in every case, even if the transaction is rejected